### ISSUE TITLE
Installing on v2.10 seems to brick device

### ISSUE THREAD

**Issue by murchu27** [dixonary/draft-reMarkable#32]
I installed draft through toltec with `opkg install draft`. After downloading and installing, I was prompted to disable the xochitl launcher, and switch to draft with:

```
systemctl disable --now xochitl
systemctl enable --now draft
```

I then rebooted the device with `reboot now`, but the device is failing to launch. It is perpetually stuck on the "reMarkable is starting" screen.

According to #31, WiFi doesn't start until xochitl does, so I can't access with SSH in order to disable draft and reenable xochitl.

Is there anything I can do here to recover the device?

**Comment by Eeems** [comment]
Are you able to ssh into the device over usb? If you are unable to ssh in over USB you'll want to give https://remarkablewiki.com/tech/recovery_rm2 a read.

I'm guessing this is a rM2 and you are running OS 2.10.3 which is not currently supported by rm2fb. https://github.com/ddvk/remarkable2-framebuffer/issues/18#issuecomment-953970979

**Comment by murchu27** [comment]
Can't reach it over USB either, the connection times out

Yes this is an rM2, v2.10.3

I'm not much of an electronics person, so I'll try the recovery route, but at the very least this has taken my tablet out of commission until I can find the parts to do this.

I know all of this is developed in people's free time, and there's no warranty, etc., of course you're not responsible for what I do with my device. But I think some due diligence needs to be applied here - no page on this repo warns that there may not be support for different versions etc., so there was no way to know that this would be unsafe.

Could the README be updated at the very least, with a marker/badge to say what versions are supported? Might save someone else from falling into the same trap as me.

**Comment by Eeems** [comment]
> Can't reach it over USB either, the connection times out

Just to confirm, you are trying to connect to the USB network's IP address when you have it connected over USB and not the wifi IP address correct? What OS are you using on your computer?

> I know all of this is developed in people's free time, and there's no warranty, etc., of course you're not responsible for what I do with my device. But I think some due diligence needs to be applied here - no page on this repo warns that there may not be support for different versions etc., so there was no way to know that this would be unsafe.
>
> Could the README be updated at the very least, with a marker/badge to say what versions are supported? Might save someone else from falling into the same trap as me.

This repo doesn't support the rm2 directly, that would be rm2fb providing a compatibility layer. As for toltec, which is what you are using for installing, which would be the real place you want to have the README/other things updated: https://github.com/toltec-dev/toltec/issues/486 https://github.com/toltec-dev/toltec/issues/445

**Comment by murchu27** [comment]
> Just to confirm, you are trying to connect to the USB network's IP address when you have it connected over USB and not the wifi IP address correct? What OS are you using on your computer?

Yep, connecting with `ssh root@10.11.99.1`, and connection times out. Whereas when I try to connect to the WiFi address, I just get "No route to host". I'm running Manjaro Linux, using OpenSSH_8.8p1


> This repo doesn't support the rm2 directly, that would be rm2fb providing a compatibility layer. As for toltec, which is what you are using for installing, which would be the real place you want to have the README/other things updated: [toltec-dev/toltec#486](https://github.com/toltec-dev/toltec/issues/486) [toltec-dev/toltec#445](https://github.com/toltec-dev/toltec/issues/445)

Okay, it seems like you are on top of this. This is frustrating, but I'm sure it's also frustrating for you. It might be worth even putting a note on this repo's README for people like myself who weren't aware of the role of rm2fb in making this work.

**Comment by Eeems** [comment]
> Yep, connecting with `ssh root@10.11.99.1`, and connection times out. Whereas when I try to connect to the WiFi address, I just get "No route to host". I'm running Manjaro Linux, using OpenSSH_8.8p1

I'd recommend rebooting your computer just in case it's messing up with routing, and then inspecting the output of `ip` for the interface if it's still failing to load. I've had Manjaro mess up every so often with the interface and rebooting resolved it.

> Okay, it seems like you are on top of this. This is frustrating, but I'm sure it's also frustrating for you. It might be worth even putting a note on this repo's README for people like myself who weren't aware of the role of rm2fb in making this work.

PRs are welcome :) Even to https://github.com/toltec-dev/web which is the website source.

**Comment by murchu27** [comment]
> I'd recommend rebooting your computer just in case it's messing up with routing, and then inspecting the output of `ip` for the interface if it's still failing to load. I've had Manjaro mess up every so often with the interface and rebooting resolved it.

Didn't help, unfortunately. I don't think the interface is showing up, I only have my loopback, ethernet, and wifi interfaces:

```
❯ ssh root@10.11.99.1
ssh: connect to host 10.11.99.1 port 22: Connection timed out

❯ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: enp2s0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc fq_codel state DOWN group default qlen 1000
    link/ether 50:9a:4c:cc:6b:ba brd ff:ff:ff:ff:ff:ff
3: wlp3s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether bc:a8:a6:dc:53:7a brd ff:ff:ff:ff:ff:ff
    inet 192.168.0.33/24 brd 192.168.0.255 scope global dynamic noprefixroute wlp3s0
       valid_lft 85956sec preferred_lft 85956sec
    inet6 fe80::45c4:fe6f:826:888a/64 scope link noprefixroute
       valid_lft forever preferred_lft forever
```

> PRs are welcome :) Even to https://github.com/toltec-dev/web which is the website source.

I'll open a PR for this repo, sure - once I order the parts for recovering :cry:

**Comment by Eeems** [comment]
> Didn't help, unfortunately. I don't think the interface is showing up, I only have my loopback, ethernet, and wifi interfaces:
> 
> ```
> ❯ ssh root@10.11.99.1
> ssh: connect to host 10.11.99.1 port 22: Connection timed out
> 
> ❯ ip a
> 1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
>     link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
>     inet 127.0.0.1/8 scope host lo
>        valid_lft forever preferred_lft forever
>     inet6 ::1/128 scope host
>        valid_lft forever preferred_lft forever
> 2: enp2s0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc fq_codel state DOWN group default qlen 1000
>     link/ether 50:9a:4c:cc:6b:ba brd ff:ff:ff:ff:ff:ff
> 3: wlp3s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
>     link/ether bc:a8:a6:dc:53:7a brd ff:ff:ff:ff:ff:ff
>     inet 192.168.0.33/24 brd 192.168.0.255 scope global dynamic noprefixroute wlp3s0
>        valid_lft 85956sec preferred_lft 85956sec
>     inet6 fe80::45c4:fe6f:826:888a/64 scope link noprefixroute
>        valid_lft forever preferred_lft forever
> ```

Looks like it's failing to create the usb network interface. Can you confirm that the USB device is showing up in dmesg? You probably don't have to recover if you can get SSH over USB working.

**Comment by murchu27** [comment]
> Looks like it's failing to create the usb network interface. Can you confirm that the USB device is showing up in dmesg? You probably don't have to recover if you can get SSH over USB working.

For absolutely no apparent reason, after having plugged in and out about a dozen times, the interface finally got created the last time that I plugged in. I'm in via SSH, and have switched back to xochitl.

I will definitely open that PR now :sweat_smile: 

Thanks for all your help, you saved my ass

TL;DR for anyone seeing this issue - don't install draft on v2.10.3 of the rM2!

**Comment by Eeems** [comment]
Glad you got it sorted! You might want to try to use configuration from https://github.com/ddvk/remarkable2-framebuffer/issues/18#issuecomment-964655946 to see if it gets rm2fb working for you, and report back on the thread if it does. If it's working an 2.10.3 is out of beta, a PR to update the default configuration would probably be in order.

> TL;DR for anyone seeing this issue - don't install draft on v2.10.3 of the rM2!

Unless you have up to date signatures that work properly for rm2fb.
